src/db.js

```diff
@@ -498,12 +498,10 @@
             return;
         }
 
-        for (var objectStoreKey in db.objectStoreNames) {
-            if (db.objectStoreNames.hasOwnProperty(objectStoreKey)) {
-                var name = db.objectStoreNames[objectStoreKey];
-                if (schema.hasOwnProperty(name) === false) {
-                    e.currentTarget.transaction.db.deleteObjectStore(name);
-                }
+        for (var i = 0; i < db.objectStoreNames.length; i++) {
+            var name = db.objectStoreNames[i];
+            if (schema.hasOwnProperty(name) === false) {
```

Comment:
Since `hasOwnProperty` is always going to be either true or false, I think it is sufficient and more common to just use the following for the `hasOwnProperty` check:

```js
if (!schema.hasOwnProperty(name)) {
```